src/db.js

```diff
@@ -204,82 +193,107 @@
 
                 return {
                     desc,
+                    distinct,
                     execute,
                     filter,
-                    distinct,
+                    limit,
                     map
                 };
             };
-            const filter = function (...args) {
-                filters.push(args.slice(0, 2));
+
+            const limit = function (...args) {
```

Comment:
Yes, agreed. I think I was just converting from `arguments` and hadn't taken it the step further as it should have, Fixed for `limit` and `filter` in the amended version I just rebased.